src/wallet/rpc/spend.cpp

```diff
@@ -1168,6 +1181,229 @@ RPCHelpMan send()
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Must provide at least one address without a specified amount");
+            }
+
+            CCoinControl coin_control;
+
+            SetFeeEstimateMode(*pwallet, coin_control, options["conf_target"], options["estimate_mode"], options["fee_rate"], /*override_min_fee=*/false);
+
+            coin_control.fAllowWatchOnly = ParseIncludeWatchonly(options["include_watching"], *pwallet);
+
+            const bool rbf{options.exists("replaceable") ? options["replaceable"].get_bool() : pwallet->m_signal_rbf};
+
+            FeeCalculation fee_calc_out;
+            CFeeRate fee_rate = GetMinimumFeeRate(*pwallet, coin_control, &fee_calc_out);
+            // Do not, ever, assume that it's fine to change the fee rate if the user has explicitly
+            // provided one
+            if (coin_control.m_feerate && fee_rate > *coin_control.m_feerate) {
+               throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Fee rate (%s) is lower than the minimum fee rate setting (%s)", coin_control.m_feerate->ToString(FeeEstimateMode::SAT_VB), fee_rate.ToString(FeeEstimateMode::SAT_VB)));
+            }
+            if (fee_calc_out.reason == FeeReason::FALLBACK && !pwallet->m_allow_fallback_fee) {
+                // eventually allow a fallback fee
+                throw JSONRPCError(RPC_WALLET_ERROR, "Fee estimation failed. Fallbackfee is disabled. Wait a few blocks or enable -fallbackfee.");
+            }
+
+            CMutableTransaction rawTx = ConstructTransaction(options["inputs"], receiver_key_value_pairs, options["locktime"], rbf);
+            LOCK(pwallet->cs_wallet);
+            std::vector<COutput> all_the_utxos;
+
+            CAmount total_input_value(0);
+            bool send_max{options.exists("send_max") && options["send_max"].get_bool()};
+            if (options.exists("inputs") && options.exists("send_max")) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot combine send_max with specific inputs.");
+            } else if (options.exists("inputs")) {
+                for (const CTxIn& input : rawTx.vin) {
+                    if (pwallet->IsSpent(input.prevout.hash, input.prevout.n)) {
+                        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Input not available. UTXO (%s:%d) was already spent.", input.prevout.hash.ToString(), input.prevout.n));
+                    }
+                    const CWalletTx* tx = pwallet->GetWalletTx(input.prevout.hash);
+                    if (!tx) {
+                        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Input not found. UTXO (%s:%d) is not part of wallet.", input.prevout.hash.ToString(), input.prevout.n));
+                    }
+                    total_input_value += tx->tx->vout[input.prevout.n].nValue;
+                }
+            } else {
+                AvailableCoins(*pwallet, all_the_utxos, &coin_control, /*nMinimumAmount=*/0);
```

Comment:
Given that `AvailableCoins` filters much more than based on effective value, I'm wondering what the expected behavior *should* be for these types of UTXOs:

- UTXOs that are not yet final or are immature coinbases 
- UTXOs for which the address has been marked as already used
- outputs from replacement transactions
etc.

Maybe it's fine that these are excluded, but just want to clarify that this behavior is what we want. Also leads me to wonder if the RPC should be returning a list of UTXOs that weren't included in the constructed transaction or something, but maybe that's overkill.

Reply:
This RPC will only operate on confirmed spendable UTXOs. It also respects the wallet-wide `avoid_reuse` configuration.

> * UTXOs that are not yet final or are immature coinbases

I would expect them to get ignored since they're not spendable at this time.

> * UTXOs for which the address has been marked as already used

If address reuse has been disabled, it should ignore funds on previously used addresses, if it only permits spending all of the funds on one address at once, it should either use all or none, and if it's not restricted, it should just spend either all or the economically viable UTXOs as determined by the options on this RPC.

> * outputs from replacement transactions

We don't spend unconfirmed UTXOs in this call.